### PR TITLE
group by argument support for pair_count

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Supported Group By argument for pair_count

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/pair_count.R [hunk 1]
```diff
@@ -6,21 +6,65 @@ pair_count <- function (df,
                         distinct = TRUE,
                         diag = FALSE,
                         sort = TRUE,
-                        unite = FALSE){
+                        unite = FALSE,
+                        group_by = NULL){
+  if(!is.null(group_by)) {
+    df <- df %>% dplyr::group_by(!!!rlang::sym(group_by))
```
Comment (hideaki) [inline R/pair_count.R:12]: Should be !! rather than !!!

Comment (hideaki) [reply R/pair_count.R:12]: Here is how to take column without quotation.
group_by_col <- tidyselect::vars_select(names(df), !! rlang::enquo(group_by))
df <- df %>% dplyr::group_by(!!!rlang::sym(group_by_col))

File: R/pair_count.R [hunk 2]
```diff
@@ -6,21 +6,65 @@ pair_count <- function (df,
...
+                         sort = FALSE,
+                         unite = FALSE) {
+  validate_empty_data(df)
+  grouped_col <- exploratory::grouped_by(df)
+  if(!is.null(grouped_col) && length(grouped_col) > 0){
+    if(group_col %in% grouped_col){
+      stop(paste0(group_col, " is grouped. Please ungroup it."))
+    }
+    # If the data frame is grouped, perform the pair_count for each group.
+    dplyr::group_modify(df, .f = function(.x, ...){
```
Comment (hideaki) [inline R/pair_count.R:45]: Probably we can call group_modify unconditionally.

File: R/pair_count.R (hunk 1 above)
Comment (hidekoji) [reply R/pair_count.R]: Changed to use `group_by_col <- col_name(substitute(group_by))` to align with other arguments.

File: R/pair_count.R (hunk 2 above)
Comment (hidekoji) [reply R/pair_count.R]: Changed to always call group_modify.

File: R/pair_count.R [hunk 3]
```diff
@@ -6,58 +6,76 @@ pair_count <- function (df,
...
               value_col,
               distinct = distinct,
               diag = diag,
               sort = sort,
-              unite = unite)
+              unite = unite,
+              group_by_col = group_by_col)
 }
 
-#' Clone implementation of now deprecated tidytext::pair_count_ for backward compatibility in Exploratory.
```
Comment (hideaki) [inline R/pair_count.R:21]: This comment should be kept.

Comment (hidekoji) [reply R/pair_count.R:21]: Brought by the comment.

**Pull request merged**